tests/unit/cluster-hostnames.tcl

```diff
@@ -0,0 +1,225 @@
+# Isolate a node from the cluster and give it a new nodeid
+proc isolate_node {id} {
+    set node_id [R $id CLUSTER MYID]
+    R $id CLUSTER RESET HARD
+    for {set j 0} {$j < [llength $::servers]} {incr j} {
+        if { $j eq $id } {
+            continue
+        }
+        R $j CLUSTER FORGET $node_id
+    }
+}
+
+# Check if cluster's view of hostnames is consistent
+proc are_hostnames_propagated {match_string} {
+    for {set j 0} {$j < [llength $::servers]} {incr j} {
+        set cfg [R $j cluster slots]
+        foreach node $cfg {
+            for {set i 2} {$i < [llength $node]} {incr i} {
+                if {! [string match $match_string [lindex [lindex [lindex $node $i] 3] 1]] } {
+                    return 0
+                }
+            }
+        }
+    }
+    return 1
+}
+
+proc get_slot_field {slot_output shard_id node_id attrib_id} {
+    return [lindex [lindex [lindex $slot_output $shard_id] $node_id] $attrib_id]
+}
+
+# Start a cluster with 3 primaries and 4 replicas. 
+start_cluster 3 4 {tags {external:skip cluster}} {
+    test "Set cluster hostnames and verify they are propagated" {
+        for {set j 0} {$j < [llength $::servers]} {incr j} {
+            R $j config set cluster-announce-hostname "host-$j.com"
+        }
+        
+        wait_for_condition 50 100 {
+            [are_hostnames_propagated "host-*.com"] eq 1
```

Comment:
isn't `are_hostnames_propagated` a subset of `wait_for_cluster_propagation`?

Reply:
No, are_hostnames_propagted also checks that the hostname is in the config. Everyone could agree on the same config but it might not have the hostname.